components/pkg-export-docker/src/build.rs

```diff
@@ -441,12 +428,23 @@ impl BuildRootContext {
         }
 
         tdeps.dedup();
-        let environment = tdeps.into_iter()
-                               .filter_map(|dep| {
-                                   let key = format!("HAB_{}", dep.to_uppercase());
-                                   env::var(&key).ok().map(|v| (key, v))
-                               })
-                               .collect();
+        let mut environment: HashMap<String, String> =
+            tdeps.into_iter()
+                 .filter_map(|dep| {
+                     let key = format!("HAB_{}", dep.to_uppercase());
+                     env::var(&key).ok().map(|v| (key, v))
+                 })
+                 .collect();
+
+        let evp = match license::env_var_present() {
+            Ok(l) => l,
+            Err(_) => false,
+        };
+
+        if evp || license::license_exists() {
```

Comment:
Based on the name, I would've expected this to be simply
```rust
if license::env_var_present() || license::license_exists() {
```
Which points to the fact that `env_var_present` is a kind of inaccurate name because it has the side effect of writing the license file and may fail. This could be simplified to:
```suggestion
        if license::env_var_present().unwrap_or(false) || license::license_exists() {
```
to eliminate the previous `match` for assigning `evp`, but I still think `env_var_present` should be renamed to something like `license::accept_if_env_var`. Also, it seems like it would be better to check `license_exists` first to avoid unnecessary failures:
```suggestion
        if license::env_var_present().unwrap_or(false) || license::license_exists() {
```
Or perhaps instead of making `env_var_present` public, there should be a `check_for_license_acceptance` or `check_for_license_acceptance_no_prompt`. That and `check_for_license_acceptance_and_prompt` could share the existing
```rust
    if license_exists() || env_var_present()? {
        return Ok(());
    }
```
logic and the `_and_prompt` version could just add the interactive bit.

Reply:
Good call, I'll make a few changes here.